src/subcommand/miner/metrics.rs

```diff
@@ -7,114 +7,126 @@ pub(crate) struct Metrics {
 }
 
 impl Metrics {
-    pub fn new() -> Self {
+    pub(crate) fn new() -> Self {
         Self {
             hashes: AtomicU64::new(0),
             shares: AtomicU64::new(0),
             started: Instant::now(),
         }
     }
 
-    pub fn add_hashes(&self, hashes: u64) {
+    pub(crate) fn add_hashes(&self, hashes: u64) {
```

Comment:
Responsive hashrate display step 3: Replace `add_hashes()` and add `current_hashrate`()`
```suggestion
    pub(crate) fn add_hashes(&self, hashes: u64) {
            self.hashes.fetch_add(hashes, Ordering::Relaxed);
            if let Ok(mut rate) = self.hashrate.lock() {
                rate.record(hashes as f64, Instant::now());
            }
    }

    pub(crate) fn current_hashrate(&self) -> f64 {
        self.hashrate.lock().map(|r| r.value()).unwrap_or(0.0)
    }
```